hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -289,6 +293,84 @@ private void sendTaskFailed(ExecutorDriver driver, Task task) {
       .setState(TaskState.TASK_FAILED)
       .build());
   }
+  
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    String taskIdStr = task.getTaskInfo().getTaskId().getValue();
+    log.info("Performing health check for task: " + taskIdStr);
+
+    int healthCheckPort = getHealthCheckPort(taskIdStr);
+    boolean taskHealthy = false;
+
+    if (healthCheckPort != -1) {
+      Socket socket = null;
+      try {
+        // TODO (elingg) with better process supervision, check which process is
+        // bound to the port.
+        // Also, possibly do a http check for the name node UI as an additional
+        // health check.
+        String localhostAddress = InetAddress.getLocalHost().getHostAddress();
+        socket = new Socket();
+        socket.bind(new InetSocketAddress(localhostAddress, healthCheckPort));
+      } catch (IOException e) {
+        taskHealthy = true;
+        log.info("Could not bind to port " + healthCheckPort + ", port is in use as expected.");
+      } catch (SecurityException | IllegalArgumentException e) {
+        log.error("Exception determining if a port is in use: ", e);
+      } finally {
+        if (socket != null) {
+          try {
+            socket.close();
+          } catch (IOException e) {
+            log.error("Error closing socket in health check: ", e);
+          }
+        }
+      }
+    }
+    
+    if (!taskHealthy) {
+      log.fatal("Node health check failed for task: " + taskIdStr);
+      killTask(driver, task.getTaskInfo().getTaskId());
+      // TODO (elingg) with better process supervision
+      // (i.e. monitoring of ZKFC's), we do not need to exit the executors
+      shutdownExecutor(2, "Failed health check");
+    }
+  }
+  
+  private int getHealthCheckPort(String taskIdStr) {
+    int healthCheckPort = -1;
+
+    if (taskIdStr.contains(HDFSConstants.DATA_NODE_ID)) {
+      healthCheckPort = HDFSConstants.DATA_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.JOURNAL_NODE_ID)) {
+      healthCheckPort = HDFSConstants.JOURNAL_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.ZKFC_NODE_ID)) {
+      healthCheckPort = HDFSConstants.ZKFC_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.NAME_NODE_ID)) {
+      healthCheckPort = HDFSConstants.NAME_NODE_PORT;
+    } else {
+      log.error("Task unknown: " + taskIdStr);
```

Comment:
we are starting to see this pattern in a number of places... I think after you commit I will refactor a number of these to a hashmap of config objects.  It will look cleaner IMO.